front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -384,13 +368,25 @@ const importTimetable = async (
   };
 
   let trainrunSectionId = 0;
+  const nodesByOpId = new Map<string, NodeDto>();
   const trainrunSections: TrainrunSectionDto[] = trainSchedules
     .map((trainSchedule) => {
       // Figure out the node ID for each path item
       const pathNodeIds = trainSchedule.path.map((pathItem) => {
         const op = findOpFromPathItem(pathItem, searchResults);
         if (op) {
-          return nodesByOpId.get(op.obj_id)!.id;
+          let node = nodesByOpId.get(op.obj_id);
+          if (!node) {
+            node = createNode({
+              trigram: op.trigram + (op.ch ? `/${op.ch}` : ''),
```

Comment:
Yeah, this logic won't be able to make the difference. But last time I asked around, `ch == ''` turned out to be a myth - there are no such OPs in the infra in practice (France or otherwise).

Note, this is old code which just happens to be moved around, it's not new code.

Reply:
Wow the ghost ch code :D Ok fine by me :)